File: Lab2/notes-app-react-native/src/services/database-service.js
import { Databases, Query } from "appwrite";
import client from "./appwrite-config";
import { APPWRITE_DATABASE_ID, APPWRITE_COLLECTION_ID } from "@env";

// Initialize the Databases SDK
const databases = new Databases(client);

// List all documents/notes in the collection
export const listDocuments = async (queries = []) => {
  try {
    const response = await databases.listDocuments(
      APPWRITE_DATABASE_ID,
      APPWRITE_COLLECTION_ID,
      queries
    );
    return response.documents;
  } catch (error) {
    console.error("Error listing documents:", error);
    throw error;
  }
};

File: Lab2/notes_app_flutter/lib/services/database_service.dart
import 'package:appwrite/appwrite.dart';
import 'package:appwrite/models.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'appwrite_config.dart';

class DatabaseService {
  final Client _client = getClient();
  late final Databases _databases;

  DatabaseService() {
    _databases = Databases(_client);
  }

  Future<List<Document>> listDocuments({List<String>? queries}) async {
    try {
      final response = await _databases.listDocuments(
        databaseId: dotenv.env['APPWRITE_DATABASE_ID']!,
        collectionId: dotenv.env['APPWRITE_COLLECTION_ID']!,
        queries: queries,
      );
      return response.documents;
    } catch (e) {
      print('Error listing documents: $e');
      throw e;
    }
  }
}